src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java

```diff
@@ -401,7 +401,7 @@ private static ValueDescriptor createField(Field field) {
                     Class<?> ct = returnType.getComponentType();
                     if (Annotation.class.isAssignableFrom(ct) && ct.getAnnotation(Repeatable.class) != null) {
                         Object res = m.invoke(a, new Object[0]);
-                        if (res != null && Annotation[].class.isAssignableFrom(res.getClass())) {
+                        if (res instanceof Annotation[]) {
                             for (Annotation rep : (Annotation[]) m.invoke(a, new Object[0])) {
```

Comment:
BTW it looks suspicious to have `m.invoke(a, new Object[0])` called again here. Shouldn't it just reuse `res` variable instead?

Reply:
It looks unnecessary. Please feel free to fix.

Reply:
Or even use the new pattern matching syntax:
```java
Object res = m.invoke(a, new Object[0]);
if (res instanceof Annotation[] anns) {
    for (Annotation rep : anns) {
```